packages/grafana-data/src/events/EventBus.ts

```diff
@@ -0,0 +1,82 @@
+import EventEmitter from 'eventemitter3';
+import { Unsubscribable, Observable } from 'rxjs';
+import { AppEvent } from './types';
+import { EventBus, LegacyEmitter, BusEventHandler, BusEventType, LegacyEventHandler, BusEvent } from './types';
+
+export class EventBusSrv implements EventBus, LegacyEmitter {
+  private emitter: EventEmitter;
+
+  constructor() {
+    this.emitter = new EventEmitter();
+  }
+
+  $emit<T extends BusEvent>(event: T): void {
+    this.emitter.emit(event.type, event);
+  }
+
+  $on<T extends BusEvent>(typeFilter: BusEventType<T>, handler: BusEventHandler<T>): Unsubscribable {
```

Comment:
Can we avoid the $ conflict by just picking new names?  Perhaps:

$emit > broadcast, publish, push?
$on > listen, handle, when?

Reply:
$emit -> publish
$on -> subscribe

@hugohaggmark thoughts?

Reply:
@torkelo I like it a lot!! 🎉